tiny_dnn/layers/average_pooling_layer.h

```diff
@@ -88,20 +89,26 @@ inline void tiny_average_pooling_back_kernel(
       float_t diff{0};
 
       for (auto connection : connections)
-        diff += prev_out[connection.first] * curr_delta[connection.second];
+        diff += prev_out.host_at(connection.first) * curr_delta.host_at(connection.second);
 
-      dW[i] += diff * scale_factor;
+      dW.host_at(i) += diff * scale_factor;
     }
 
     for (size_t i = 0; i < bias2out.size(); i++) {
       const std::vector<size_t> &outs = bias2out[i];
       float_t diff{0};
 
-      for (auto o : outs) diff += curr_delta[o];
+      for (auto o : outs) diff += curr_delta.host_at(o);
 
-      db[i] += diff;
+      db.host_at(i) += diff;
     }
+    (*in_grad[1])[sample] = dW.toVec();
+    (*in_grad[2])[sample] = db.toVec();
+    (*in_grad[0])[sample] = prev_delta.toVec();
+    (*out_grad[0])[sample] = curr_delta.toVec();
```

Comment:
I understand all these ``toVec`` won't be necessary in the future because everything will be a Xtensor. Am I right?